### DISCUSSION TITLE
npm 7 destroys itself if update is cancelled early

### DISCUSSION THREAD

**Discussion by targos** [npm/feedback#116] [CLI] [answered]
Because npm 7 is not latest yet, I have to remember to add the version number in my command to update it: `npm i -g npm@7`.
If I forget to do it, realize that shortly after launching the command, and stop the update with CTRL+C, npm doesn't work anymore (it cannot be found in the PATH). This is annoying because the only way to fix the broken state in my case is to switch to another Node.js version, and to delete and reinstall the broken one.
I know there has to be a moment during which the old version is actually replaced by the newly installed one, but the window can probably be made much shorter. npm 6 is not affected by this issue.

**Comment by MylesBorins** [comment, marked as answer]
Hey @targos nice to see you

This probably makes a bit more sense to open as an issue over on https://github.com/npm/cli/issues would you be up for opening something up there?

**Reply by targos** [reply]
Done: https://github.com/npm/cli/issues/2312

Sorry for opening at the wrong place.
